src/lib/operators/table_scan.cpp

```diff
@@ -94,7 +94,7 @@ std::shared_ptr<const Table> TableScan::_on_execute() {
   for (ChunkID chunk_id{0u}; chunk_id < in_table->chunk_count(); ++chunk_id) {
     if (excluded_chunk_set.count(chunk_id)) continue;
 
-    auto job_task = std::make_shared<JobTask>([=, &output_mutex, &output_chunks]() {
+    auto job_task = std::make_shared<JobTask>([this, in_table, chunk_id, &output_mutex, &output_chunks]() {
```

Comment:
Why did you decide to copy the shared_ptr `in_table`?

Reply:
Before, `in_table` was passed by value via `=`.  Now, we could either 

* redeclare `const auto in_table = input_table_left();` or 
* use `input_table_left()` 

inside the Lambda. What do you suggest?

Reply:
Why can't we capture it by reference? That should have been done before, too, but now that we explicitly mention it, it looks like we gave it some thought.

Reply:
Ok, I thought this was by done with intention. But since we have `CurrentScheduler::wait_for_tasks(jobs)` at the end of the function, it should also work with `&in_table`.